src/main/java/com/mraof/minestuck/entity/underling/BasiliskEntity.java

```diff
@@ -1,151 +1,182 @@
+                .add(Attributes.KNOCKBACK_RESISTANCE, 0.6).add(Attributes.MOVEMENT_SPEED, 0.25)
+                .add(Attributes.ATTACK_DAMAGE, 6);
+    }
+
+    protected SoundEvent getAmbientSound() {
+        return MSSoundEvents.ENTITY_BASILISK_AMBIENT;
+    }
+
+    protected SoundEvent getHurtSound(DamageSource damageSourceIn) {
+        return MSSoundEvents.ENTITY_BASILISK_HURT;
+    }
+
+    protected SoundEvent getDeathSound() {
+        return MSSoundEvents.ENTITY_BASILISK_DEATH;
+    }
+
+    @Override
+    public GristSet getGristSpoils() {
+        return GristHelper.generateUnderlingGristDrops(this, damageMap, 6);
+    }
+
+    @Override
+    protected int getVitalityGel() {
+        return random.nextInt(3) + 4;
+    }
+
+    @Override
+    protected void onGristTypeUpdated(GristType type) {
+        super.onGristTypeUpdated(type);
+        applyGristModifier(Attributes.MAX_HEALTH, 20 * type.getPower(), AttributeModifier.Operation.ADDITION);
+        applyGristModifier(Attributes.ATTACK_DAMAGE, 2.7 * type.getPower(), AttributeModifier.Operation.ADDITION);
+        this.xpReward = (int) (6 * type.getPower() + 4);
+    }
+
+    @Override
+    public void die(DamageSource cause) {
+        super.die(cause);
+        Entity entity = cause.getEntity();
+        if (this.dead && !this.level.isClientSide) {
+            computePlayerProgress((int) (30 + 2.4 * getGristType().getPower())); //most basilisks stop giving xp at rung 32
+            firstKillBonus(entity, (byte) (Echeladder.UNDERLING_BONUS_OFFSET + 2));
+        }
+    }
+
+    @Override
+    public boolean canBeCollidedWith() {
+        return false;
+    }
+
+    public void checkDespawn() {
+    }
+
+    @Override
+    public boolean isPickable() {
+        return false;
+    }
+
+    @Override
+    public void aiStep() {
+        super.aiStep();
+        Vector3d[] positions = new Vector3d[this.parts.length];
+
+        for(int j = 0; j < this.parts.length; ++j) {
+            positions[j] = new Vector3d(this.parts[j].getX(), this.parts[j].getY(), this.parts[j].getZ());
+        }
+
+        float bodyAngle= this.yBodyRot * ((float)Math.PI / 180F);
+        float xOffset = MathHelper.sin(bodyAngle);
+        float zOffset = -MathHelper.cos(bodyAngle);
+
+        this.updatePart(this.body, xOffset * 0.5, 0, zOffset * 0.5);
+        this.updatePart(this.head, xOffset * -2.5, 0.3, zOffset * -2.5);
+        this.updatePart(this.tail, xOffset * 2.5, 0, zOffset * 2.5);
+        this.updatePart(this.tailEnd, xOffset * 4.5, 1, zOffset * 4.5);
+
+        for(int l = 0; l < this.parts.length; ++l) {
+            this.parts[l].xo = positions[l].x;
+            this.parts[l].yo = positions[l].y;
+            this.parts[l].zo = positions[l].z;
+            this.parts[l].xOld = positions[l].x;
+            this.parts[l].yOld = positions[l].y;
+            this.parts[l].zOld = positions[l].z;
+        }
+    }
```

Comment:
This would be an exceptionally good area to add comments to, so that future improvements to multipart entities are done with a better understanding of what came before